packages/online/src/useOnline.ts

```diff
@@ -16,7 +19,7 @@ function getIsOnline(): boolean {
  * @returns {boolean} The value of navigator.onLine
  */
 function useOnline(): boolean {
-  const [online, changeOnline] = useState<boolean>(false);
+  const [online, changeOnline] = useState<boolean>(getIsOnline());
```

Comment:
Here, at each render, getIsOnline function will be called. You should pass a callback function to useState. This callback will only be called once.

```js
const [online, changeOnline] = useState<boolean>(() => getIsOnline());
```

Reply:
useState with callback? That would mean that the `online` initial value will be a function. Right?

Reply:
We can actually do something like this,

```javascript
const onlineInitial = useMemo(() => getIsOnline(), [])
const [online, changeOnline] = useState(onlineInitial);
```

Reply:
No. We have the possibility to pass a callback function to useState ([lazy initial state](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state)). React will call this callback function only once.

> "The initialState argument is the state used during the initial render. In subsequent renders, it is disregarded."

The result returned by this callback function will be the new state. 

I made an example: [https://stackblitz.com/edit/react-xs2bu2](https://stackblitz.com/edit/react-xs2bu2)

It's the best way to set an initial state value from a function. It's not the same that
```js
const [online, changeOnline] = useState<boolean>(getIsOnline());
```
In this case, at each render, getIsOnline function will be called. What we want is to have only an initial value from a function call on the first render.

Reply:
Nice. Let's do this ? @maciekgrzybek

Reply:
@thibaultboursier nice one didn't know about it :)